### ISSUE TITLE
Ability to work with subclasses of perfmetrics.StatsDClient

### ISSUE THREAD

**Issue by cutz** [NextThought/nti.fakestatsd#17]
The ``FakeStatsDClient`` get the bulk of its functionality/interface by subclasses perfmetrics' ``StatsDClient``. If someone was using a custom subclass of ``StatsDClient`` there isn't a great way to use that implementation with this package. Should that be easier?

**Comment by jamadden** [comment]
Yeah, easier is better, if we think this might be a common thing to do. 

This could be a place that "constructor is a factory" (plus interfaces) could help us, e.g., by turning the "class" into a constructor that optionally takes a class to subclass, if needed.
